momento-redis/hash_commands_test.go

```diff
@@ -0,0 +1,148 @@
+package momento_redis_test
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/momentohq/client-sdk-go/auth"
+	"github.com/momentohq/client-sdk-go/config"
+	"github.com/momentohq/client-sdk-go/momento"
+	momentoredis "github.com/momentohq/momento-go-redis-client/momento-redis"
+
+	. "github.com/momentohq/momento-go-redis-client/momento-redis/test_helpers"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/redis/go-redis/v9"
+)
+
+var _ = Describe("Dictionary methods", func() {
+	sContext := NewSharedContext()
+	BeforeEach(func() {
+		cacheName := fmt.Sprintf("golang-redis-%s", uuid.NewString())
+		switch sContext.UseRedis {
+		case true:
+			host := "127.0.0.1"
+			port := "6379"
+			sContext.Client = redis.NewClient(&redis.Options{
+				Addr: host + ":" + port,
+			})
+		case false:
+			credential, _ := auth.NewEnvMomentoTokenProvider(AuthTokenEnvVariable)
+			mClient, _ := momento.NewCacheClient(config.LaptopLatest(), credential, 60*time.Second)
+			sContext.MomentoClient = mClient
+			// create cache; it resumes execution normally incase the cache already exists and isn't exceptional
+			_, createErr := sContext.CreateCache(sContext.Ctx, mClient, cacheName)
+			if createErr != nil {
+				panic("Failed to create cache with cache name " + cacheName + "\n" + createErr.Error())
+			}
+
+			sContext.Client = momentoredis.NewMomentoRedisClient(mClient, cacheName)
+		}
+		DeferCleanup(func() {
+			if sContext.UseRedis {
+				sContext.Client.FlushDB(sContext.Ctx)
+			} else {
+				_, deleteErr := sContext.DeleteCache(sContext.Ctx, sContext.MomentoClient, cacheName)
+				if deleteErr != nil {
+					panic("Failed to delete cache with cache name " + cacheName + "\n" + deleteErr.Error())
+				}
+			}
+		})
+	})
```

Comment:
this looks fine for now but if we start repeating this in too many places it might be a good idea to see if we can centralize it?